sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -8,6 +8,17 @@
 #include <string>
 #include <thread>
 
+// Macro for setting libcurl handle options
+#define LIBCURL_SET_OPT(HANDLE, OPT, VALUE, ERROR) \
```

Comment:
any reason to not make this a function?

Reply:
somehow I though I need the macro to return. But since it just throws, making a function is better yes.
Tks

Reply:
make an inline function inside unnamed namespace - it is the best option here.

Reply:
```
namespace
{
    inline void ThrowIfCurlSetoptFailed(handle, opt, value, std::string& errorMsg)
    {
        ...
}
```

Reply:
oh wait. I need VALUE to accept string or int and I don't think creating another struct for a Union or a template is worth it